src/os/unix/unix.cpp

```diff
@@ -151,8 +152,8 @@ std::string OTTD2FS(std::string_view name)
 {
 	static iconv_t convd = (iconv_t)(-1);
 	if (convd == (iconv_t)(-1)) {
-		const char *env = GetLocalCode();
-		convd = iconv_open(env, INTERNALCODE);
+		auto env = GetLocalCode();
+		convd = iconv_open(env.data(), INTERNALCODE);
```

Comment:
Maybe better keep `GetLocalCode` returning `char *`, so it's clear it's NUL terminated.

Reply:
That'd 'upstream' to `GetCurrentLocale` as well, so the suggestion is to drop the 'locales' commit from this PR. Right?

An alternative would be dropping iconv support for non-MacOS. Although I wonder what we'd need iconv for when we actually use `std::filesystem::path` instead of `std::string`.

Reply:
`iconv_open` is called only once. So you can as well create a `std::string` here.

Reply:
That's a good compromise.